### ISSUE TITLE
Link to xml route doesn't work in exported site

### ISSUE THREAD

**Issue by altano** [vercel/next.js#52778]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
❯ pnpm next info

    Operating System:
      Platform: linux
      Arch: x64
      Version: Ubuntu 20.04.0 LTS Mon Jul 17 2023 03:32:35 GMT-0700 (Pacific Daylight Time)
    Binaries:
      Node: 16.20.0
      npm: 9.4.2
      Yarn: 1.22.19
      pnpm: 8.6.3
    Relevant Packages:
      next: 13.4.10-canary.8
      eslint-config-next: 13.4.10
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
    Next.js Config:
      output: export


warn  - Latest canary version not detected, detected: "13.4.10-canary.8", newest: "13.4.10".
        Please try the latest canary version (`npm install next@canary`) to confirm the issue still exists before creating a new issue.
        Read more - https://nextjs.org/docs/messages/opening-an-issue
```

### Link to the code that reproduces this issue or a replay of the bug

https://stackblitz.com/edit/stackblitz-starters-hr381n?file=src%2Fapp%2Frss.xml%2Froute.ts

### To Reproduce

Wait till the site spins up. A link to "RSS Feed rss.xml" will show up. click it, get a 404 error

<img width="1237" alt="image" src="https://github.com/vercel/next.js/assets/1009/0f17bdc2-4284-4f5f-a873-67ca5f3088e6">

### Describe the Bug

I've created an rss feed served out of ./src/app/rss.xml/route.ts

This feed works correctly if I:
✅ navigate to /rss.xml in dev mode
✅ navigate to / and click a link to rss.xml in dev mode
✅ build the site and `next start` (export:undefined), navigate to /, click the link to rss.xml
✅ export the site (export:true, build), serve it, and navigate to /rss.xml

It does NOT work if I:
❌ export the site, serve it using serve package, navigate to /, click the link to rss.xml

I see two failures in the console/network pane:
1. On the initial page load of / I get a 404 hitting `https://stackblitzstartersyc6jaw-15xy--3000--c53ab388.local-credentialless.webcontainer.io/rss.xml.txt?_rsc=1ag7k`
2. When clicking the link to "/rss.xml" I get a 404 as the browser tries to navigate to "/rss" (no extension)

I verified in the dev tools that the anchor looks correct: `<a href="/rss.xml">RSS Feed rss.xml</a>` so I believe there is some rogue JS click handler that is causing a bad navigation.

alternatively: [repro with next@canary](https://stackblitz.com/edit/stackblitz-starters-yc6jaw?file=src%2Fapp%2Frss.xml%2Froute.ts)

### Expected Behavior

I should be able to create links to my rss.xml route in the exported version of the site.

### Which browser are you using? (if relevant)

Chrome & Firefox tested, same result.

**Comment by icyJoseph** [comment]
Hi @altano this has been fixed in next@13.4.11 https://stackblitz.com/edit/stackblitz-starters-znbgdj?file=package.json

**Comment by altano** [comment]
Confirmed, thank you for the heads up, @icyJoseph !
